Use the correct pressure attributes

judahrand opened the pull request:
I'm not sure what the reasons for the choices were for the previous state but I've tried these changes and they work well for me.

The boiler fill still results in water being forced out of the grouphead confirming the boiler in indeed full when the steam wand is closed. But when the steam wand is open it runs until timeout. To me this suggests that this still achieves the desired effect but without having to wait for the `smoothedPressure` to reach the threshold. I could be wrong here and maybe more testing is needed to stop using `smoothedPressure` here... but for me at least (using the standard BoM) it isn't sensitive enough and doesn't behave reliably.

For the `StopConditions`, this seems more logical? We don't want to trigger the condition due to noise in the transducer. We only want to transition on steady state pressure, right?

Together these changes make both the `Boiler Fill` and `Switch on Threshold` behave reliably across a range of transducer hose lengths (20-60cm). I'm yet to try it with a capillary tube but I suspect that's irrelevant given it isn't standard BoM.

judahrand commented:
I've investigated the boiler fill portion of this change some more and it looks to me like changing it to `.pressure` is the 'correct' change. It seems as though it did in fact used to be `.smoothedPressure` but was inadvertently changed in [this](https://github.com/Zer0-bit/gaggiuino/commit/568dfff4e2cf7c49a8e580fc949c08106a08016e) commit. I believe when this was merged was when I originally started having issues with the boiler fill.

Zer0-bit commented:
The stop conditions is an unfortunate omission that slipped through when the switch to a externalised library model was made, that's smth that would have been found, though you might want to look into your transducer setup as it looks like it's noisy af as on my end even with this "raw" pressure output it was behaving as intended which shows how important is having a proper setup.

Regarding the boiler fill it's wrong and i specifically went away from marking the boiler as full just based on raw pressure.
To me it seems like you don't understand why this change was made and what boiler fill was done for.

PR closed as it contains unwanted changes.

judahrand commented:
> that's smth that would have been found

It was found. It was like that since the beginning of December and now it is found. I appreciate that it was just a mistake.

> though you might want to look into your transducer setup as it looks like it's noisy af as on my end even with this "raw" pressure output it was behaving as intended which shows how important is having a proper setup.

I've looked at everything I can think of other than the boiler & grouphead itself! I've replaced the pump, the transducer, I've tried various lengths of hose to the transducer, I've moved to PCB V3. Clearly others are having some similar issues (both excessive boiler filling and 'switch on threshold' triggering early) as there are at least two `bugfix` and/or `install-help` threads.

> Regarding the boiler fill it's wrong and i specifically went away from marking the boiler as full just based on raw pressure.
To me it seems like you don't understand why this change was made and what boiler fill was done for.

I'm curious as to what problem you were aiming to solve by moving to `smoothedPressure` for the boiler fill, yes. My impression is the the boiler fill algo works by detecting the restriction increase when water is forced down the internal. The purpose of this being so that there is always a consistent volume of water in the boiler before brew - this is important for temperature consistency / control as well as predictive scales (since otherwise the assumption that all water that is pumped after a brew starts hits the puck is broken). Am I misunderstanding? If so would you be willing to share your understanding with me?
